probe-rs/src/probe/arm_jtag.rs

```diff
@@ -824,9 +822,12 @@ fn parse_swd_response(response: &[bool], direction: TransferDirection) -> Result
 }
 
 pub trait RawProtocolIo {
-    fn jtag_io<M, I>(&mut self, tms: M, tdi: I) -> Result<Vec<bool>, DebugProbeError>
+    fn jtag_shift_tms<M>(&mut self, tms: M, tdi: bool) -> Result<(), DebugProbeError>
```

Comment:
While we are at it, could we add some docs here and also make it mandatory for the file? :)